rcl/test/rcl/test_log_level.cpp

```diff
@@ -0,0 +1,206 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "rcl/rcl.h"
+#include "rcl/log_level.h"
+#include "rcl/error_handling.h"
+#include "rcutils/logging.h"
+
+#include "./arg_macros.hpp"
+
+#ifdef RMW_IMPLEMENTATION
+# define CLASSNAME_(NAME, SUFFIX) NAME ## __ ## SUFFIX
+# define CLASSNAME(NAME, SUFFIX) CLASSNAME_(NAME, SUFFIX)
+#else
+# define CLASSNAME(NAME, SUFFIX) NAME
+#endif
+
+class CLASSNAME (TestLogLevelFixture, RMW_IMPLEMENTATION) : public ::testing::Test
+{
+};
+
+#define EXPECT_INVALID_RET_FOR_ARGUMENTS_LOG_LEVEL(...) \
+  { \
+    rcl_ret_t ret; \
+    rcl_arguments_t local_arguments = rcl_get_zero_initialized_arguments(); \
+    const char * local_argv[] = {__VA_ARGS__}; \
+    unsigned int local_argc = (sizeof(local_argv) / sizeof(const char *)); \
+    ret = rcl_parse_arguments( \
+      local_argc, local_argv, rcl_get_default_allocator(), &local_arguments); \
+    ASSERT_EQ(RCL_RET_INVALID_ROS_ARGS, ret); \
+  }
+
+TEST(CLASSNAME(TestLogLevelFixture, RMW_IMPLEMENTATION), error_log_level) {
+  EXPECT_INVALID_RET_FOR_ARGUMENTS_LOG_LEVEL(
+    "process_name", "--ros-args", "--log-level",
+    "=debug");
```

Comment:
I think we usually avoid using `=` in command line arguments, because `bash` or other shells might use them for something else.
We tend to use `:=` in its place.

I'm not sure if using `=` here is ok, or if we should use `:=`.